citrination_client/data/client.py

```diff
@@ -0,0 +1,267 @@
+        :param glob: A regex used to select one or more files in the dataset
+        :param version_number: The version number of the dataset to retrieve files from
+        :return: A list of :class:`DatasetFile` objects matching the provided pattern.
+        """
+        if version_number is None:
+            latest = True
+        else:
+            latest = False
+
+        data = {
+            "download_request": {
+                "glob": glob,
+                "isDir": is_dir,
+                "latest": latest
+            }
+        }
+
+        response = self._post_json(routes.matched_files(dataset_id), data)
+        failure_message = "Failed to get matched files in dataset {}".format(dataset_id)
+        versions = http_utils.get_success_json(response, failure_message)['versions']
+
+        if version_number is None:
+            version = versions[0]
+        else:
+            version = list(filter(lambda v: v['number'] == version_number, versions))
+
+        return list(
+            map(
+                lambda f: DatasetFile(path=f['filename'], url=f['url']), version['files']
+                )
+            )
+
+    def get_dataset_file(self, dataset_id, file_path, version = None):
+        """
+        Retrieves a dataset file matching a provided file path
+
+        :param data_set_id: The id of the dataset to retrieve file from
+        :param file_path: The file path within the dataset
+        :param version: The dataset version to look for the file in. If nothing is supplied, the latest dataset version will be searched
+        :return: A :class:`DatasetFile` object matching the filepath provided
+        """
+        if version == None:
+            result = self._get(routes.file_dataset_path(dataset_id, file_path))
+        else:
+            result = self._get(routes.file_dataset_version_path(dataset_id, version, file_path))
+
+        file = http_utils.get_success_json(result, "An error occurred retrieving file {}".format(file_path))['file']
+
+        return DatasetFile(path=file['filename'], url=file['url'])
+
+    def get_pif(self, dataset_id, uid, version = None):
+        """
+        Retrieves JSON representation of a PIF from a given dataset.
+
+        :param data_set_id: The id of the dataset to retrieve PIF from
+        :param uid: The uid of the PIF to retrieve
+        :param version: The dataset version to look for the PIF in. If nothing is supplied, the latest dataset version will be searched
+        :return: A :class:`Pif` object
+        """
+        if version == None:
+            response = self._get(routes.pif_dataset_uid(dataset_id, uid))
+        else:
+            response = self._get(routes.pif_dataset_version_uid(dataset_id, uid, version))
+
+        response = http_utils.check_success(response, "An error occurred retrieving PIF {}".format(uid))
+        return pif.loads(response.content)
+
+    def create_dataset(self, name=None, description=None, public=False):
```

Comment:
in this method and the following method, handle the "0"/"1" case without advertising it (comment #utahisrad)

Reply:
👍